sql/core/src/test/scala/org/apache/spark/sql/errors/QueryCompilationErrorsSuite.scala

```diff
@@ -964,6 +964,35 @@ class QueryCompilationErrorsSuite
         "className" -> "org.apache.spark.sql.catalyst.expressions.UnsafeRow"))
   }
 
+  test("SPARK-47102: Collation in CollateContext when COLLATION_ENABLED is false") {
+    withSQLConf(SQLConf.COLLATION_ENABLED.key -> "false") {
+      checkError(
+        exception = intercept[AnalysisException] {
+          sql(s"select 'aaa' collate 'UNICODE_CI'")
```

Comment:
how about the type parsing like `CREATE TABLE t(col STRING COLLATE ...)`?

Reply:
+1 on adding such a test.